src/gleam/result.gleam

```diff
@@ -366,6 +367,60 @@ pub fn all(results: List(Result(a, e))) -> Result(List(a), e) {
   list.try_map(results, fn(x) { x })
 }
 
+/// Iterates through a list of results and returns the first success, if any.
+/// Returns the list otherwise, which will thus be a list of errors.
+/// 
+/// ## Examples
+/// 
+/// ```gleam
+/// > any([Error("e"), Ok(1), Error("2")])
+/// Ok(1)
+/// ```
+/// 
+/// ```gleam
+/// > any([Error("e"), Error("b"), Error("c")])
+/// [Error("e"), Error("b"), Error("c")]
+/// ```
+pub fn any(results: List(Result(a, e))) -> Result(a, List(e)) {
+  case list.find_map(results, function.identity) {
+    Ok(result) -> Ok(result)
+    Error(_) -> results |> list.map(fn(err) {
+      case err {
+        Error(e) -> e
+        _ -> panic
+      }
+    }) |> Error
+  }
+}
+
+/// Iterates through a list of result-returning functions and returns the first success, if any.
+/// Returns the list of errors otherwise.
```

Comment:
I feel like the documentation should be more explicit about what happens when the first success is met: a user may be interested in knowing wether it short-circuits or keeps executing all the other functions in the list